Added inheritance support for Exiled.Permissions

jeppevinkel opened the pull request:
Added inheritance support for the permissions.
It works by adding the name of any groups to inherit from in the `Inheritance` list.

iopietro commented on Exiled.Permissions/Features/Group.cs:
```diff
@@ -31,5 +38,25 @@ public class Group
+        /// Gets the combined permissions of the group plus all inherited groups.
+        /// </summary>
+        [YamlIgnore]
+        public List<string> CombinedPermissions
```
Shouldn't we set this Property only when `Permissions.Reload()` get called?

jeppevinkel replied:
The current solution essentially does that, but it might be better to explicitly do it.

jeppevinkel commented:
I've changed it to all be done in `Permissions.Reload()` now.

I also added support for nested inheritance in descending order like this:
```yml
owner:
  inheritance:
  - admin
admin:
  inheritance:
  -  moderator
moderator:
  inheritance: []
```
It only works in that order, not the other way.
To make it work all ways would be more complicated, time consuming, and could result in infinite loops if not careful.
Currently if someone does it the wrong way, they simply don't get the permissions.
